src/content/__tests__/monetizationLinkManager.test.ts

```diff
@@ -0,0 +1,569 @@
+import { JSDOM } from 'jsdom';
+import { MonetizationLinkManager } from '@/content/services/monetizationLinkManager';
+import type {
+  ContentToBackgroundMessage,
+  MessageManager,
+  Response,
+} from '@/shared/messages';
+import { success, failure } from '@/shared/helpers';
+import type { Logger } from '@/shared/logger';
+import type { WalletAddress } from '@interledger/open-payments';
+
+// for syntax highlighting
+const html = String.raw;
+
+const WALLET_ADDRESS: WalletAddress[] = [
+  {
+    authServer: 'https://auth.example.com',
+    publicName: 'Test Wallet USD',
+    assetCode: 'USD',
+    assetScale: 2,
+    id: '',
```

Comment:
Let's add more real-ish values here. Then for wallet addresses, we can use `WALLET_ADDRESSES[0].id` instead of hard-coding values multiple times in tests.